### PR TITLE
feat: Lesson07branch main file is updated completely. 

### PR DESCRIPTION
**Pull request by javyW**

### CODE REVIEW

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if (price >= 15.00) {
+    System.out.println("Don't buy book (Too expensive)");
+    } else {
+        System.out.println("Buy the book (Good price to buy)");
+    }
+}
+
+
+// One of your functions must make use of a loop.
+   public static void main(String[] args) {
```
Comment (HeirJordan3) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:83]: you do not need a main method in your class here.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
+package com.codedifferently.lesson7;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:1]: Also, the package name should be `com.codedifferently.lesson7.javyenware` (all lowercase). You should update the folder name as well to lowercase (here and for your test file).

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Configuration;
+
+
+// Design at least one custom class that represents a real-world object.
+public class Book {
+
+   // (Javyens comment) The extends keyword indicates that the class is inherited from the book class
+   public static class NoBookAvailable extends Exception {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:13]: Use the exception somewhere and add a test ensuring that it is used correctly.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+       public NoBookAvailable (string insertMessage) {
+        super(insertMessage);
+       }
+   }
+
+
+// The class must have at least 5 member variables of at least 3 different types.
+   private string Title;
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:21]: Member variables should be lower camel case, please refactor.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+
+// You must also incorporate an enum type as well.
+   private enum Genre {
+       NONFICTION, FICTION, HORROR,
+       MYSTERY, FANTASY, HISTORY;
+   }
+
+
+// The class must have at least one constructor.
+public Book(string Title, string Author, string Publisher, short YearReleased, float Price, byte PageNumber) {
```
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:37]: Parameter names should be lower camel case.

File: lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java (hunk 1 above)
Comment (anthonydmays) [reply lesson_07/objects/objects_app/src/main/java/com/codedifferently/lesson7/JavyenWare/Book.java:83]: Probably easiest to add something like a collection of author names and use that for your example.

File: lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/JavyenWaretest/BookTest.java
Comment (anthonydmays) [inline lesson_07/objects/objects_app/src/test/java/com/codedifferently/lesson7/JavyenWaretest/BookTest.java:1]: Moar tests please.

**Review by anthonydmays — CHANGES_REQUESTED** [review]

**Pull request closed without merging**